Add installation documentation

mauritsvanrees opened the pull request:
As promised during the previous Plone 6 meeting, and done within the last hour before the next meeting. :-)

I am not sure if I will be able to make it to this meeting, still need to get a car, a daughter, and possibly a son before that. :-D

This is an edited copy of what I included in the recent release notes.  See:
https://dist.plone.org/release/6.0.0a4/RELEASE-NOTES.md

For the moment it includes a section on how to install Volto. This is a candidate for removal because the Volto docs are included automatically and I point to it. But the Volto docs also talk about how to install a backend. So maybe keep for now and improve later.

I called the main file `installation.md`. I am fine with renaming this to `installing.md` or `install.md` if that sounds more natural to a native speaker. I will want to point to this file from the new release notes, so best pick a name now and stick with it.

Note: this is my first PR here with MyST, so I am still a rookie on that count. Patience and guidance are welcome. :-)

mauritsvanrees committed:
Add backend installation documentation.

This is an edited copy of what I included in the recent release notes.  See:
https://dist.plone.org/release/6.0.0a4/RELEASE-NOTES.md

For the moment it includes a section on how to install Volto.
This is a candidate for removal because the Volto docs are included automatically and I point to it.
But the Volto docs also talk about how to install a backend.
So maybe keep for now and improve later.

I called the main file `installation.md`.
I am fine with renaming this to `installing.md` or `install.md` if that sounds more natural to a native speaker.
I will want to point to this file from the new release notes, so best pick a name now and stick with it.

stevepiercy commented:
I'll complete my review later tonight. But for now...

We have many ways to install Plone—or parts of Plone such as Volto, backend, and Classic UI—but all of them lack context. We need to add context. For example:

------

## Development

### Backend only

If you want to install, develop, test, or write documentation for _Plone backend only_, then use the following instructions.

### Volto only

If you want to install, develop, test, or write documentation for _Volto only_, then use the following instructions.


## Production

If you want to _deploy Plone 6 to a production environment_, then use the following instructions.

------

I lack a full understanding of how all the Plone pieces fit together, but I think the above examples describe what I seek. In other words, the Plone Installation docs should be an entry point for all users which then point them in the right direction.

stevepiercy commented:
I looked at this more closely. I realized that we need a better place for installation documentation, and any other item that does not fit into any of the current primary navigation items. We currently have:

- [Frontend](https://6.dev-docs.plone.org/volto/index.html)
- [REST API](https://6.dev-docs.plone.org/plone.restapi/docs/source/index.html)
- [Backend](https://6.dev-docs.plone.org/backend/index.html)
- [Classic UI](https://6.dev-docs.plone.org/classic-ui/index.html)
- [Contributing to Plone Documentation](https://6.dev-docs.plone.org/contributing/index.html)

I firmly believe that "Install" belongs in the primary navigation. Would you please create a new directory `install`, rename the file to `index.md`, move it into the new directory, and finally update the `toctree` in `/index.md`? It should be the first item in the `toctree`.

stevepiercy commented:
Also this is more than just backend installation. You're talking about Plone 6 "the whole thing" Installation. That's why I think it belongs at a top navigation level.

mauritsvanrees committed:
Make install doc its own main chapter.

mauritsvanrees commented:
> I firmly believe that "Install" belongs in the primary navigation.

That makes sense.

> Would you please create a new directory `install`, rename the file to `index.md`, move it into the new directory, and finally update the `toctree` in `/index.md`? It should be the first item in the `toctree`.

Done.

stevepiercy committed:
Overhaul installation documentation, based on @mauritsvanrees's initial contribution.
This version provides context, structure, and order, leading the developer down the path to a successful installation experience according to their specific needs.
This is still a work in progress. Additional review is needed.

stevepiercy commented:
@mauritsvanrees thank you! I decided I would take a stab as well, and pushed a commit with an overhaul, based on your initial contribution.

This version is a work in progress. It needs review from several core contributors. Who else should review this? @sneridagh @tiberiuichim @jensens @ericof others?

I approached this with the view of "I am new to Plone 6. How do I install it?" I think that includes pretty much anyone who is not a core Plone package developer at this point. 😉  That audience probably does not know the differences between a Docker and manual installation, or choosing Classic UI or Volto for a fronted.

Finally I firmly believe that we must have one, and only one, place for Plone installation documentation. As such I opened a [new issue with Volto](https://github.com/plone/volto/issues/3289). I hope to resolve discrepancies and maintenance issues there. Each package should have its own documentation for _development_ purposes, such as for running tests, and that should be separate from _installation_.

jensens commented on docs/install/index.md:
```diff
@@ -0,0 +1,253 @@
+# Install Plone 6 with constrained requirements with pip
+bin/pip install Plone -c https://dist.plone.org/release/6.0.0a4/constraints.txt
+# Create a Plone 6 site with the given username and password in the current directory
+bin/mkwsgiinstance -u admin:admin -d .
```
mkwsgiinstance is not enough, like no blobs configured. I would like to point to cookiecutter-zope-instance here.

jensens replied:
this one https://github.com/bluedynamics/cookiecutter-zope-instance/
I can move it to zopefoundation or plone repos if this feels better.

stevepiercy replied:
Yeah, and the ensuing warning is... not a good look. I will trust you and other core developers to update this as you feel is best.

jensens replied:
Well, we have two sprints now, and I am at both. This is part of our installation story, so stay tuned. As said, mkwsgiinstance wont make anybody happy.

stevepiercy replied:
FYI, I will also remotely participate in both upcoming sprints. I would be happy to act as a rubber duck.

mauritsvanrees commented:
@stevepiercy Your update LGTM.

ericof reviewed (approved):
LGTM, let's merge it and built on top of it
